src/Microsoft.Android.Sdk.Analysis/ResourceDesignerDiagnosticSuppressor.cs

```diff
@@ -0,0 +1,71 @@
+using System.Linq;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class ResourceDesignerDiagnosticSuppressor : DiagnosticSuppressor
```

Comment:
When I was writing this:
* https://github.com/jonathanpeppers/memory-analyzers

The project template for a Roslyn analyzer created a unit test project:
* https://github.com/jonathanpeppers/memory-analyzers/tree/main/MemoryAnalyzers/MemoryAnalyzers.Test

Would the same test project work here? We could do it in a future PR.

I'm thinking there are other things we could add to this in the future like:
* https://github.com/dotnet/android/wiki/Potential-Roslyn-Analyzers

Reply:
I think something like that for a future PR would be good. And this new project is a good place for analyzers as well as suppressors.